### DISCUSSION TITLE
在 Windows QtWidgets 中，使用子线程离屏渲染帧崩溃

### DISCUSSION THREAD

**Discussion by KouBro** [Tencent/libpag#2319] [Q&A] [answered]
如果在显示 Qt 窗口的情况下，在子线程中线程离屏渲染帧会崩溃。
如果在主线程中离屏渲染帧，或者在子线程中离屏渲染帧但是不显示窗口的情况下，都是正常工作。
![image](https://github.com/Tencent/libpag/assets/47996669/0c45e173-05e6-4e8f-92ce-a51fa1b09631)
请问pag不能再子线程中使用吗，还是需要做一些同步的操作？

**Comment by domchen** [comment, marked as answer]
QT平台的OpenGL环境限制非常多，参考根目录的QT 文件夹下的demo，那是唯一可以正常子线程渲染到窗口的方法。如果你用的还是老的QWidget不是QtQuick框架，那就没办法渲染在子线程。

**Reply by KouBro** [reply]
了解了，感谢解答👍
